src/qt/pivx/masternodewizarddialog.cpp

```diff
@@ -242,116 +252,122 @@ bool MasterNodeWizardDialog::createMN()
-                fwrite(lineCopy.c_str(), std::strlen(lineCopy.c_str()), 1, configFile);
-                fclose(configFile);
-
-                fs::path pathOldConfFile("old_masternode.conf");
-                if (!pathOldConfFile.is_complete()) pathOldConfFile = GetDataDir() / pathOldConfFile;
-                if (fs::exists(pathOldConfFile)) {
-                    fs::remove(pathOldConfFile);
-                }
-                rename(pathMasternodeConfigFile, pathOldConfFile);
-
-                fs::path pathNewConfFile("masternode.conf");
-                if (!pathNewConfFile.is_complete()) pathNewConfFile = GetDataDir() / pathNewConfFile;
-                rename(pathConfigFile, pathNewConfFile);
-
-                mnEntry = masternodeConfig.add(alias, ipAddress+":"+port, mnKeyString, txID, indexOutStr);
-
-                // Lock collateral output
-                COutPoint collateralOut(walletTx->GetHash(), indexOut);
-                walletModel->lockCoin(collateralOut);
-
-                returnStr = tr("Master node created! Wait %1 confirmations before starting it.").arg(MASTERNODE_MIN_CONFIRMATIONS);
-                return true;
-            } else {
-                returnStr = tr("masternode.conf file doesn't exists");
+        if (iss >> comment) {
+            if (comment.at(0) == '#') continue;
+            iss.str(line);
+            iss.clear();
+        }
+
+        if (!(iss >> alias >> ip >> privKey >> txHash >> outputIndex)) {
+            iss.str(line);
+            iss.clear();
+            if (!(iss >> alias >> ip >> privKey >> txHash >> outputIndex)) {
+                streamConfig.close();
+                returnStr = tr("Error parsing masternode.conf file");
+                return false;
             }
-        } else {
-            returnStr = tr("Cannot send collateral transaction.\n\n%1").arg(returnMsg);
         }
+        lineCopy += line + "\n";
+    }
+
+    if (lineCopy.size() == 0) {
+        lineCopy = "# Masternode config file\n"
+                   "# Format: alias IP:port masternodeprivkey collateral_output_txid collateral_output_index\n"
+                   "# Example: mn1 127.0.0.2:51472 93HaYBVUCYjEMeeH1Y4sBGLALQZE1Yc1K64xiqgX37tGBDQL8Xg 2bcd3c84c84f87eaa86e4e56834c92927a07f9e18718810b92e0d0324456a67c 0"
+                   "#";
+    }
+    lineCopy += "\n";
+
+    streamConfig.close();
+
+    std::string txID = collateralOut.hash.ToString();
+    std::string indexOutStr = std::to_string(collateralOut.n);
+
+    // Check IP address type
+    QHostAddress hostAddress(addressStr);
+    QAbstractSocket::NetworkLayerProtocol layerProtocol = hostAddress.protocol();
+    if (layerProtocol == QAbstractSocket::IPv6Protocol) {
+        ipAddress = "["+ipAddress+"]";
     }
-    return false;
+
+    fs::path pathConfigFile("masternode_temp.conf");
+    if (!pathConfigFile.is_complete()) pathConfigFile = GetDataDir() / pathConfigFile;
+    FILE* configFile = fopen(pathConfigFile.string().c_str(), "w");
```

Comment:
nit: use `fsbridge::fopen`

Reply:
will address this in a follow up PR which refactors the update of the masternode.conf out of `MasterNodeWizardDialog::createMN()`